AboghanemAO/lab1/Source.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <conio.h>
+using namespace std;
+
+
+
+int main() {
+
+    float a;
+    float b;
+    char operation_sign;
+    bool right_input = false;
+    char ch;
+
+
+
+    do
+    {
+        while (!right_input)
+        {
+            cout << "Enter two numbers and an operator sign in this format (number1 number2 operation_sign) : ";
+
+            cin >> a >> b;
+            while (std::cin.fail()) {
+                std::cout << "Wrong input try again\n" << "Enter two numbers and an operator sign in this format (number1 number2 operation_sign)  ";
+                std::cin.clear();
+                std::cin.ignore(256, '\n');
+                std::cin >> a >> b;
+            }
+
+            cin >> operation_sign;
+
+
+
+            switch (operation_sign)
+            {
+
+            case '+':
+                std::cout << "Your answer is : " << a + b << endl;
```

Comment:
Please extract line 
`std::cout << "Your answer is : " << a ? b << endl;` from switch..case in order to reduce code duplication

Reply:
fixed
but i have a question :
now when the operation sign is wrong the output will be
"Your answer is : Wrong operation sign try again"
should i add (if statement) before switch ? so if the the operation sign is wrong it doesn't print "Your answer is :"

Reply:
Yes, it is the proper code flow